modules/hostserv/offer.c

```diff
@@ -252,6 +252,27 @@ myuser_is_in_group(struct myuser *mu, struct myentity *mt)
 	return false;
 }
 
+static void
+take_vhost(struct sourceinfo *si, const char *host)
+{
+	char local_host[BUFSIZE];
+	if (strstr(host, "$account"))
+	{
+		mowgli_strlcpy(local_host, host, HOSTLEN);
```

Comment:
Not quite; `BUFSIZE`. The third parameter is the size of the destination; it will already terminate early upon copying a null character.